src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceLifetime.cs

```diff
@@ -116,6 +125,7 @@ protected override void OnStart(string[] args)
         /// <remarks>This might be called multiple times by service Stop, ApplicationStopping, and StopAsync. That's okay because StopApplication uses a CancellationTokenSource and prevents any recursion.</remarks>
```

Comment:
This comment is now wrong and in fact was a signal that there was an issue here.

I fixed the extra call from `StopAsync`, the call from `ApplicationStopping` existed at [the time of this comment](https://github.com/dotnet/extensions/commit/05d1a6eb0e051a0cbadc64246f77fe9b836d33a6#diff-e4d0d3a9aa655704afc92739fc3402911ce12676b9e316d14f45cbc16f3fca4eR38) but was removed [later](https://github.com/dotnet/extensions/commit/43435a80d007d568aa17d891ea32ca87363716fc#diff-899fddd5f7ff286e68af676af2b391e6e284d41a2c02ed3e1678c997dd843e06L38).  At this point `OnStop` should only ever be called once.